ocs_ci/utility/utils.py

```
@@ -4113,12 +4111,24 @@ def get_latest_ds_olm_tag(upgrade=False, latest_tag=None, stable_upgrade_version
         upgrade = False
+    tags = get_ocs_olm_operator_tags(filter_tag_name=f"like:{ocs_version}")
+    latest_image = None
     for tag in tags:
         if tag["name"] == latest_tag:
             latest_image = tag["manifest_digest"]
             break
     if not latest_image:
-        raise TagNotFoundException("Couldn't find latest tag!")
+        log.warning(
+            f"Tag '{latest_tag}' not found in version-filtered results, "
+            "fetching all tags"
```

Ruff (0.15.21)

[warning] Logging statement uses f-string

(G004)